src/main/java/de/teamlapen/vampirism/tileentity/AlchemicalCauldronTileEntity.java

```diff
@@ -1,514 +1,282 @@
-        Fluid fluid = stack.getFluid();
-//        if (fluid.equals(FluidRegistry.WATER)) { //TODO 1.14
-//            return 0xC03040FF;
-//        } else if (fluid.equals(FluidRegistry.LAVA)) {
-//            return 0xFFFF5010;
-//        }
-        int color = fluid.getColor(stack);
-        if (color == 0xFFFFFFFF) {
-            color = 0xFFFFFFFE; //0xFFFFFFFF == -1 which makes our isLiquidCheck fail
+    @Override
+    public void tick() {
+        boolean flag = this.isBurning();
+        boolean flag1 = false;
+        if (flag) {
+            this.field_214013_b.set(0, this.field_214013_b.get(0) - 1);
         }
-        return color;
-    }
 
-    /**
-     * @return f the given stack is a (registed) liquid stack
-     */
-    private static boolean isLiquidStack(ItemStack stack) {
-        return getLiquidColor(stack) != -1;
-    }
+        if (!this.world.isRemote) {
+            ItemStack itemstack = this.items.get(1);
+            if (this.isBurning() || !itemstack.isEmpty() && !this.items.get(3).isEmpty() && !this.items.get(0).isEmpty()) {
+                AlchemicalCauldronRecipe irecipe = this.world.getRecipeManager().getRecipe((IRecipeType<AlchemicalCauldronRecipe>) this.recipeType, this, this.world).orElse(null);
+                if (!this.isBurning() && this.func_214008_b(irecipe) && this.canPlayerCook(irecipe)) {
+                    field_214013_b.set(0, this.getBurnTime(itemstack));
+                    field_214013_b.set(1, field_214013_b.get(0));
+                    if (this.isBurning()) {
+                        flag1 = true;
+                        if (itemstack.hasContainerItem())
+                            this.items.set(1, itemstack.getContainerItem());
+                        else if (!itemstack.isEmpty()) {
+                            Item item = itemstack.getItem();
+                            itemstack.shrink(1);
+                            if (itemstack.isEmpty()) {
+                                this.items.set(1, itemstack.getContainerItem());
+                            }
+                        }
+                    }
+                }
 
-    private int totalBurnTime = 0, burnTime = 0, cookTime = 0, totalCookTime = 0;
-    private boolean cookingClient;
-    private boolean burningClient;
-    private int liquidColor;
-    @OnlyIn(Dist.CLIENT)
-    private ISoundReference boilingSound;
-    /**
-     * Can contain a recipe which can be cooked by the owner.
-     * Is set when a {@link IAlchemicalCauldronRecipe#canBeCooked(int, ISkillHandler)} check succeeds.
-     * As long as the current recipe is equals to this one, no new check is executed.
-     * Thereby the owner can e.g. disconnect and the cauldron can continue cooking (as long as the recipe does not change).
-     */
-    @Nullable
-    private IAlchemicalCauldronRecipe checkedRecipe;
+                if (this.isBurning() && this.func_214008_b(irecipe) && this.canPlayerCook(irecipe)) {
```

Comment:
When using SRG names feel free to add a comment saying what they might do, so once get mapped it is easier to identify what they do.
If this is just a plain copy of vanilla, add a reference instead